src/runtime/async.js

```diff
@@ -22,11 +143,51 @@ function observeForEach(observe, next) {
         reject(error);
       },
       return(value) {
-        resolve();
+        resolve(value);
       }
     });
   });
 }
 
+function schedule(asyncF) {
+  return Promise.resolve().then(asyncF);
+}
+
+const generator = Symbol();
+const onDone = Symbol();
+
+class DecoratedGenerator {
+  constructor(_generator, _onDone) {
+    this[generator] = _generator;
+    this[onDone] = _onDone;
+  }
+
+  next(value) {
+    var result = this[generator].next(value);
+    if (result !== void 0 && result.done) {
```

Comment:
We haven't been very strict about `void 0` vs `undefined` but please be consistent. Previously, in this file you used `undefined`.